Update macOS installation instructions for OpenCV for OR Connector (#279)
Vantiq/vantiq-extension-sources: objectRecognitionSource/README.md

*   [ImageRetrieverInterface](#retrieveInterface) -- An interface that allows different image retrieval mechanisms to be
        more easily integrated without changes to the rest of the code.
    *   [CameraRetriever](#camRet) -- Retrieves images from a directly connected camera using OpenCV<sup>&trade;</sup>.
    *   [NetworkStreamRetriever](#netRet) -- Retrieves images from an IP camera.
    *   [FileRetriever](#fileRet) -- Retrieves images and videos from disk.
    *   [FtpRetriever](#ftpRet) -- Retrieves images through FTP, FTPS, and SFTP.

## How to Run the Program<a name="objRecMain" id="objRecMain"></a>

### Building OpenCV <a name="building-opencv" id="building-openv"></a>

The [official site](https://opencv.org/releases/) and [the somewhat dated tutorial](https://opencv-java-tutorials.readthedocs.io/en/latest/01-installing-opencv-for-java.html) describe how to install it. These are, we believe, sufficient in Windows and Linux environments.

#### Building OpenCV for macOS

Installation on macOS<sup>&reg;</sup> is a bit more complicated. The easiest way is to use [Homebrew](https://brew.sh). _Homebrew_ is a package manager for macOS.  It is also available for Linux.

> Once _Homebrew_ is installed, you can read the instructions in [this slightly dated tutorial]() for background information. However, recent changes in _Homebrew_ have made some changes necessary.

Reviewer: Forgot to link to the dated tutorial.

Specfically, current versions of _Homebrew_ need a bit more information about the Java environment to build the Java code for OpenCV. If this information is not provided, the installation of OpenCV appears to succeed, but the Java implemenation is not available. No errors are apparent. 

As a result, the following steps are necessary to install OpenCV with Java support on macOS. The following are expected to be run from the Terminal application.

1. If not already installed, install _Homebrew_ (see the [installation instructions](https://brew.sh).) The _Homebrew_ command is `brew`, so you will see its use below.
2. If not already installed, install _Ant_ (used by the build process).
    *`brew install ant`.
    * Set the ANT_HOME environment variable to the installation location.  Due to some issues with the current versions of the _Ant_ installation, you may need to to set ANT_HOME to `/usr/local/Cellar/ant/1.10.11/libexec` where _1.10.11_ is the version of _Ant_ installed.
3. If not already installed, install the XCode command-line tools.
    * `xcode-select install`

> 4. Prepare to use `brew` to install OpenCV. To do this, you will configure the OpenCV installation to include the Java libraries.  Do do this, you will edit the `brew` _formula_ (the instructions `brew` uses to do the installation.

Reviewer: ```suggestion
4. Prepare to use `brew` to install OpenCV. To do this, you will configure the OpenCV installation to include the Java libraries.  This is done by editing the `brew` _formula_ (the instructions `brew` uses to do the installation.
```

Or something along those lines

Author: I think I've spent enough time in `brew` this week that "do do" was just to hard to avoid...

> 4. Prepare to use `brew` to install OpenCV. To do this, you will configure the OpenCV installation to include the Java libraries.  This is done by editing the `brew` _formula_ (the instructions `brew` uses to do the installation.

Reviewer 2: missing )
```suggestion
4. Prepare to use `brew` to install OpenCV. To do this, you will configure the OpenCV installation to include the Java libraries.  This is done by editing the `brew` _formula_ (the instructions `brew` uses to do the installation).
```

    * `brew edit opencv`
    * In the text editor that opens, make the following changes.
        * Find the line `-DBUILD_JAVA=OFF` and change it to `-DBUILD_JAVA=ON`
        * Find the line `-DBUILD_opencv_java=OFF` and change it to `-DBUILD_opencv_java=ON`
        * Below this line, add the following lines
        
        ```
            -DJAVA_INCLUDE_PATH=<JAVA_HOME>/include
            -DJAVA_INCLUDE_PATH2=<JAVA_HOME>/include/darwin
            -DJAVA_AWT_INCLUDE_PATH=<JAVA_HOME>/include
            -DOPENCV_JAVA_TARGET_VERSION=<Java runtime version>
        ```
        * In these additions,
            * Replace `<JAVA_HOME>` with the value of your `JAVA_HOME` environment variable, and
            * Replace  `<Java runtime version>` with the version of the Java runtime you use (_e.g._, 1.8).

>             * For example, on this developers machine, these lines are as follows:

Reviewer 2: ```suggestion
            * For example, on this developer's machine, these lines are as follows:
```

    *   If you know that you will not use specific classes in the *.neuralNet or *.imageRetriever packages you can
        remove them now.
    *   FileRetriever, NetworkStreamRetriever, and CameraRetriever all use OpenCV, which is not in the gradle
        dependencies. If they are not removed you must install OpenCV, set the environment variable
        OPENCV_LOC to the folder containing the jar, and ensure that when running your code the jar is in the classpath

>         and (lib)opencv_javaXXX.dll/.so/.dylib is in a folder in `java.library.path` (where XXX is replace by the version of OpenCV you've installed -- _e.g._, 453, 410, _etc_). You may see a warning about not 

Reviewer 2: replace -> replaced
```suggestion
        and (lib)opencv_javaXXX.dll/.so/.dylib is in a folder in `java.library.path` (where XXX is replaced by the version of OpenCV you've installed -- _e.g._, 453, 410, _etc_). You may see a warning about not 
```

        having (lib)opencv_javaXXX.dll/.so/.dylib in the correct directory when compiling the objectRecognition jar,
        but this can be safely ignored since you will be setting the path yourself. Instructions for installing

>         OpenCV can be found at the [official site](https://opencv.org/releases/). Please see the [Building OpenCV](#building-opencv) for instructions on generating these libraries.

Reviewer 2: remove "the" ?
```suggestion
        OpenCV can be found at the [official site](https://opencv.org/releases/). Please see [Building OpenCV](#building-opencv) for instructions on generating these libraries.
```
